src/nvim/ops.c

```diff
@@ -3439,10 +3439,12 @@ void ex_display(exarg_T *eap)
           MSG_PUTS_ATTR("^J", attr);
           n -= 2;
         }
-        for (p = yb->y_array[j]; *p && (n -= ptr2cells(p)) >= 0; ++p) {
+        n -= ptr2cells(p);
+        for (p = yb->y_array[j]; *p && n >= 0; p++) {
           clen = (*mb_ptr2len)(p);
           msg_outtrans_len(p, clen);
           p += clen - 1;
+          n -= ptr2cells(p);
```

Comment:
this would need to be guarded by `if (*p)`, as done in the old code.

Reply:
Or we can write ` n -= ptr2cells(p);` in the beginning of the for loop body, and check `n >= 0` by an if statement. Though I have nothing against ignoring it either.